From 2536101ec13dd4cfebb96f2a8dc0eb35d4148dff Mon Sep 17 00:00:00 2001
From: dergoegge <n.goeggi@gmail.com>
Date: Fri, 23 May 2025 13:45:11 +0100
Subject: [PATCH] p2p: Reintroduce blocktxn assertion crash bug

---
 src/blockencodings.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/blockencodings.cpp b/src/blockencodings.cpp
index 5f4061a71d..9a7bf648b8 100644
--- a/src/blockencodings.cpp
+++ b/src/blockencodings.cpp
@@ -182,7 +182,7 @@ bool PartiallyDownloadedBlock::IsTxAvailable(size_t index) const
 
 ReadStatus PartiallyDownloadedBlock::FillBlock(CBlock& block, const std::vector<CTransactionRef>& vtx_missing)
 {
-    if (header.IsNull()) return READ_STATUS_INVALID;
+    assert(!header.IsNull());
 
     uint256 hash = header.GetHash();
     block = header;
-- 
2.47.2

